internal/agentsession/capture.go

```
@@ -81,6 +81,382 @@ func Capture(sessionStore, cwd string, launchedAt time.Time, claimed map[string]
+// start. A resumed conversation's first turn pushes one of the first two.
+const hermesActivitySQL = `COALESCE(s.last_activity_at, (SELECT MAX(m.timestamp) FROM messages m WHERE m.session_id = s.id), s.started_at)`
+
+func queryHermesActivity(db *sql.DB, ctx context.Context, cwd string, add func(id string, sessionCwd sql.NullString, activity float64) bool) error {
+	rows, err := db.QueryContext(ctx, `
+		SELECT s.id, s.cwd, `+hermesActivitySQL+`
+		FROM sessions s
+		WHERE s.source = 'cli'`)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `rows.Close` is not checked

(errcheck)